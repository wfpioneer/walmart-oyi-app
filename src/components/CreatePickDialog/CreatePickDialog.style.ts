import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const styles = StyleSheet.create({
  closeContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    width: '100%',
    height: 45,
    alignContent: 'flex-start'
  },
  pickerContainer: {
    borderWidth: 1,
    borderRadius: 10,
    borderColor: COLOR.GREY,
    marginHorizontal: 20,
    marginBottom: 10
  },
  locationText: {
    textAlign: 'center'
  },
  numericSelectorContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
    marginBottom: 10
  },
  switchContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 20,
    marginBottom: 10,
    paddingHorizontal: 10
  },
  buttonContainer: {
    marginHorizontal: 20
  }
});

export default styles;
