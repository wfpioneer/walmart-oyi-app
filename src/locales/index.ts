import * as RNLocalize from 'react-native-localize';
import I18n from 'i18n-js';
// @ts-ignore
import moment from 'moment';
import 'moment/locale/es';
import en from './en';
import es from './es';
import zh from './zh';

export const setI18nConfig: () => void = () => {
  // fallback if no available language fits
  const fallback = { languageTag: 'en', isRTL: false };

  const { languageTag } = RNLocalize.findBestAvailableLanguage(['en', 'es', 'zh'])
    || fallback;

  // set i18n-js config
  I18n.translations = { en, es, zh };
  I18n.locale = languageTag;

  // the following is to set up moment to work with spanish localization
  moment.locale(languageTag);
};

export const setLanguage = (language: string) => {
  if (language !== 'en' && language !== 'es' && language !== 'zh') {
    throw new Error('Unsupported language');
  }

  I18n.locale = language;
  moment.locale(language);
};

export const strings: (
  tag: string,
  options?: I18n.TranslateOptions | undefined,
) => string = (tag: string, options: I18n.TranslateOptions | undefined = {}) => I18n.t(tag, options);

export const currencies: (
  value: number,
  options?: I18n.ToCurrencyOptions | undefined
) => string = (value: number, options: I18n.ToCurrencyOptions | undefined = {}) => I18n.toCurrency(value, options);

export const numbers: (
  value: number,
  options?: I18n.ToNumberOptions | undefined
) => string = (value: number, options: I18n.ToNumberOptions | undefined = {}) => I18n.toNumber(value, options);
