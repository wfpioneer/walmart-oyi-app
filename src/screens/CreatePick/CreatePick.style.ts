import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    flex: 1
  },
  itemDetailsView: {
    padding: 7,
    backgroundColor: COLOR.WHITE
  },
  pickParamView: {
    borderColor: COLOR.GREY,
    borderTopWidth: 2,
    borderBottomWidth: 2
  },
  pickLocationAddButton: {
    marginLeft: 'auto'
  },
  pickLocationAddText: {
    color: COLOR.MAIN_THEME_COLOR
  },
  pickParamLine: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 10,
    borderTopWidth: 2,
    borderColor: COLOR.GREY,
    height: 60
  },
  pickerContainer: {
    borderWidth: 1,
    borderRadius: 10,
    borderColor: COLOR.GREY,
    width: '65%'
  },
  createButtonView: {
    padding: 10,
    marginTop: 'auto'
  }
});

export default styles;
