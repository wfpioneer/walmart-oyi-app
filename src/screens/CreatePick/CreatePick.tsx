import React, { useState } from 'react';
import {
  Pressable,
  SafeAreaView,
  Text,
  View
} from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { Picker } from '@react-native-picker/picker';
import ItemInfo from '../../components/iteminfo/ItemInfo';
import NumericSelector from '../../components/NumericSelector/NumericSelector';
import Button from '../../components/buttons/Button';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import ItemDetails from '../../models/ItemDetails';
import Location from '../../models/Location';
import { strings } from '../../locales';
import styles from './CreatePick.style';

export const MOVE_TO_FRONT = 'moveToFront';
export const PALLET_MIN = 1;
export const PALLET_MAX = 99;

interface CreatePickProps {
  item: ItemDetails;
  selectedSectionState: [string, React.Dispatch<React.SetStateAction<string>>]
  palletNumberState: [number, React.Dispatch<React.SetStateAction<number>>]
  floorLocations: Location[];
}

export const CreatePickScreen = (props: CreatePickProps) => {
  const {
    item, selectedSectionState, palletNumberState, floorLocations
  } = props;

  const [selectedSection, setSelectedSection] = selectedSectionState;
  const [palletNumber, setPalletNumber] = palletNumberState;

  const pickerLocations = (locations: Location[]) => [
    <Picker.Item label={strings('PICKING.SELECT_LOCATION')} value="" key={-1} />,
    ...locations.map((location: Location) => (
      <Picker.Item label={location.locationName} value={location.locationName} key={location.locationName} />
    )),
    <Picker.Item label={strings('PICKING.MOVE_TO_FRONT')} value={MOVE_TO_FRONT} key={MOVE_TO_FRONT} />
  ];

  const onPalletIncrease = (
    numberOfPallets: number,
    setNumberOfPallets: React.Dispatch<React.SetStateAction<number>>
  ) => {
    if (numberOfPallets < PALLET_MAX) {
      setNumberOfPallets((previousState: number) => previousState + 1);
    }
  };

  const onPalletDecrease = (
    numberOfPallets: number,
    setNumberOfPallets: React.Dispatch<React.SetStateAction<number>>
  ) => {
    if (numberOfPallets > PALLET_MIN) {
      setNumberOfPallets((previousState: number) => previousState - 1);
    }
  };

  const onPalletTextChange = (text: string, setNumberOfPallets: React.Dispatch<React.SetStateAction<number>>) => {
    const newQty = parseInt(text, 10);
    if (!Number.isNaN(newQty) && newQty >= PALLET_MIN && newQty <= PALLET_MAX) {
      setNumberOfPallets(newQty);
    }
  };

  const isNumberOfPalletsValid = (numberOfPallets: number) => (
    numberOfPallets >= PALLET_MIN && numberOfPallets <= PALLET_MAX
  );

  const disableCreateButton = () => !selectedSection || !item.location.reserve;

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.itemDetailsView}>
        <ItemInfo
          category={`${item.categoryNbr} - ${item.categoryDesc}`}
          itemName={item.itemName}
          itemNbr={item.itemNbr}
          upcNbr={item.upcNbr}
          price={item.price}
          status={item.status || ''}
          hidePrintButton={true}
        />
      </View>
      <View style={styles.pickParamView}>
        <View style={styles.pickParamLine}>
          <MaterialCommunityIcons name="map-marker-outline" size={40} />
          <Text>{strings('ITEM.LOCATION')}</Text>
          <Pressable onPress={() => {}} style={styles.pickLocationAddButton}>
            <Text style={styles.pickLocationAddText}>{strings('GENERICS.ADD')}</Text>
          </Pressable>
        </View>
        <View style={styles.pickParamLine}>
          <Text>{strings('PICKING.RESERVE_LOC')}</Text>
          <Text>{item.location.reserve && item.location.reserve[0].locationName}</Text>
        </View>
        <View style={styles.pickParamLine}>
          <Text>{strings('PICKING.FLOOR_LOC')}</Text>
          <View style={styles.pickerContainer}>
            <Picker
              selectedValue={selectedSection}
              onValueChange={section => setSelectedSection(section)}
              mode="dropdown"
            >
              {pickerLocations(floorLocations)}
            </Picker>
          </View>
        </View>
        <View style={styles.pickParamLine}>
          <Text>{strings('PICKING.NUMBER_PALLETS')}</Text>
          <NumericSelector
            isValid={isNumberOfPalletsValid(palletNumber)}
            onDecreaseQty={() => onPalletDecrease(palletNumber, setPalletNumber)}
            onIncreaseQty={() => onPalletIncrease(palletNumber, setPalletNumber)}
            onTextChange={(text: string) => onPalletTextChange(text, setPalletNumber)}
            minValue={PALLET_MIN}
            maxValue={PALLET_MAX}
            value={palletNumber}
          />
        </View>
      </View>
      <View style={styles.createButtonView}>
        <Button title={strings('GENERICS.CREATE')} disabled={disableCreateButton()} />
      </View>
    </SafeAreaView>
  );
};

const CreatePick = () => {
  const picking = useTypedSelector(state => state.Picking);
  const floorLocations = useTypedSelector(state => state.ItemDetailScreen.floorLocations);
  const selectedSectionState = useState('');
  const palletNumberState = useState(1);

  const mockLocations: Location[] = [
    {
      aisleId: 2,
      aisleName: '1',
      locationName: 'ABAR1-1',
      sectionId: 3,
      sectionName: '1',
      type: 'floor',
      typeNbr: 2,
      zoneId: 1,
      zoneName: 'ABAR'
    },
    {
      aisleId: 2,
      aisleName: '2',
      locationName: 'ABAR1-2',
      sectionId: 4,
      sectionName: '2',
      type: 'floor',
      typeNbr: 2,
      zoneId: 1,
      zoneName: 'ABAR'
    }
  ];

  // May need to change this as not all item details are stored in item details redux
  const mockItem: ItemDetails = {
    categoryNbr: 73,
    itemName: 'treacle tart',
    itemNbr: 2,
    upcNbr: '8675309',
    backroomQty: 765432,
    basePrice: 4.92,
    categoryDesc: 'Deli',
    claimsOnHandQty: 765457,
    completed: false,
    consolidatedOnHandQty: 65346,
    location: {
      reserve: [
        {
          aisleId: 5018,
          aisleName: '1',
          locationName: 'ABAR1-1',
          sectionId: 5019,
          sectionName: '1',
          type: 'reserve',
          typeNbr: 1,
          zoneId: 3632,
          zoneName: 'ABAR'
        }
      ],
      count: 1
    },
    onHandsQty: 76543234,
    pendingOnHandsQty: 2984328947,
    price: 4.92,
    replenishment: {
      onOrder: 100000
    },
    sales: {
      daily: [{
        day: 'Thursday',
        value: 3
      }],
      dailyAvgSales: 500,
      lastUpdateTs: 'right now',
      weekly: [{
        week: 34,
        value: 654
      }],
      weeklyAvgSales: 3500
    }
  };

  return (
    <CreatePickScreen
      item={mockItem}
      selectedSectionState={selectedSectionState}
      palletNumberState={palletNumberState}
      floorLocations={mockLocations}
    />
  );
};

export default CreatePick;
