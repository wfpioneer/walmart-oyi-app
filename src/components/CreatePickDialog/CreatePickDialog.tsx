import React from 'react';
import { Switch, Text, View } from 'react-native';
import { Picker } from '@react-native-picker/picker';
import Button from '../buttons/Button';
import IconButton from '../buttons/IconButton';
import NumericSelector from '../NumericSelector/NumericSelector';
import styles from './CreatePickDialog.style';
import { ModalCloseIcon } from '../../screens/Modal/Modal';
import { strings } from '../../locales';
import Location from '../../models/Location';
import { MOVE_TO_FRONT, PALLET_MAX, PALLET_MIN } from '../../screens/CreatePick/CreatePick';
import COLOR from '../../themes/Color';

interface CreatePickDialogProps {
  selectedSection: string;
  setSelectedSection: React.Dispatch<React.SetStateAction<string>>;
  numberOfPallets: number;
  setNumberOfPallets: React.Dispatch<React.SetStateAction<number>>;
  isQuickPick: boolean;
  setIsQuickPick: React.Dispatch<React.SetStateAction<boolean>>;
  locations: Location[];
  onClose(): void;
  onSubmit(): void;
}

const addLocationsToPicker = (locations: Location[]) => [
  <Picker.Item label={strings('PICKING.SELECT_LOCATION')} value="" key={-1} />,
  ...locations.map((location: Location) => (
    <Picker.Item label={location.locationName} value={location.locationName} key={location.locationName} />
  )),
  <Picker.Item label={strings('PICKING.MOVE_TO_FRONT')} value={MOVE_TO_FRONT} key={MOVE_TO_FRONT} />
];

const onPalletIncrease = (
  numberOfPallets: number,
  setNumberOfPallets: React.Dispatch<React.SetStateAction<number>>
) => {
  if (numberOfPallets < PALLET_MAX) {
    setNumberOfPallets((preaviousState: number) => preaviousState + 1);
  }
};

const onPalletDecrease = (
  numberOfPallets: number,
  setNumberOfPallets: React.Dispatch<React.SetStateAction<number>>
) => {
  if (numberOfPallets > PALLET_MIN) {
    setNumberOfPallets((preaviousState: number) => preaviousState - 1);
  }
};

const onPalletTextChange = (text: string, setNumberOfPallets: React.Dispatch<React.SetStateAction<number>>) => {
  const newQty = parseInt(text, 10);
  if (!Number.isNaN(newQty) && newQty >= PALLET_MIN && newQty <= PALLET_MAX) {
    setNumberOfPallets(newQty);
  }
};

const isNumberOfPalletsValid = (numberOfPallets: number) => (
  numberOfPallets >= PALLET_MIN && numberOfPallets <= PALLET_MAX
);

const onQuickPickChange = (setIsQuickPick: React.Dispatch<React.SetStateAction<boolean>>) => (
  setIsQuickPick((previousState: boolean) => !previousState)
);

export const CreatePickDialog = (props: CreatePickDialogProps): JSX.Element => {
  const {
    selectedSection,
    setSelectedSection,
    numberOfPallets,
    setNumberOfPallets,
    locations,
    onClose,
    isQuickPick,
    setIsQuickPick,
    onSubmit
  } = props;
  return (
    <View>
      <View style={styles.closeContainer}>
        <IconButton
          icon={ModalCloseIcon}
          type={Button.Type.NO_BORDER}
          onPress={() => onClose()}
          testID="closeButton"
        />
      </View>
      <Text style={styles.locationText}>
        {strings('PICKING.SELECT_LOCATION')}
      </Text>
      <View style={styles.pickerContainer}>
        <Picker
          selectedValue={selectedSection}
          onValueChange={section => setSelectedSection(section)}
          mode="dropdown"
        >
          {addLocationsToPicker(locations)}
        </Picker>
      </View>
      {selectedSection ? (
        <View style={styles.numericSelectorContainer}>
          <Text>
            {strings('PICKING.NUMBER_PALLETS')}
          </Text>
          <NumericSelector
            isValid={isNumberOfPalletsValid(numberOfPallets)}
            onDecreaseQty={() => onPalletDecrease(numberOfPallets, setNumberOfPallets)}
            onIncreaseQty={() => onPalletIncrease(numberOfPallets, setNumberOfPallets)}
            onTextChange={(text: string) => onPalletTextChange(text, setNumberOfPallets)}
            minValue={PALLET_MIN}
            maxValue={PALLET_MAX}
            value={numberOfPallets}
          />
        </View>
      ) : null}
      <View style={styles.switchContainer}>
        <Text style={styles.quickPickText}>
          {strings('PICKING.QUICK_PICK')}
        </Text>
        <Switch
          value={isQuickPick}
          onValueChange={() => onQuickPickChange(setIsQuickPick)}
          thumbColor={COLOR.MAIN_THEME_COLOR}
          trackColor={{ false: COLOR.GREY, true: COLOR.MAIN_THEME_COLOR }}
          testID="quickPickSwitch"
        />
      </View>
      <View style={styles.buttonContainer}>
        <Button
          disabled={selectedSection === ''}
          title={strings('PICKING.CREATE_PICK')}
          type={Button.Type.PRIMARY}
          onPress={onSubmit}
          testID="submitButton"
        />
      </View>
    </View>
  );
};

export default CreatePickDialog;
