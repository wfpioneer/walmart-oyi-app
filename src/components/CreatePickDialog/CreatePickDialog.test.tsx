import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CreatePickDialog } from './CreatePickDialog';
import { MOVE_TO_FRONT } from '../../screens/CreatePick/CreatePick';
import { strings } from '../../locales';

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'mockMaterialCommunityIcons');

describe('Test createPickDialog Component', () => {
  it('Renders the createPickDialog component with location other than moveToFront selected', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <CreatePickDialog
        selectedSection=""
        setSelectedSection={jest.fn()}
        numberOfPallets={1}
        setNumberOfPallets={jest.fn()}
        isQuickPick={false}
        setIsQuickPick={jest.fn()}
        onClose={jest.fn()}
        onSubmit={jest.fn()}
        locations={[]}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('Renders the createPickDialog component with location moveToFront selected', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <CreatePickDialog
        selectedSection={MOVE_TO_FRONT}
        setSelectedSection={jest.fn()}
        numberOfPallets={1}
        setNumberOfPallets={jest.fn()}
        isQuickPick={false}
        setIsQuickPick={jest.fn()}
        onClose={jest.fn()}
        onSubmit={jest.fn()}
        locations={[]}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('test buttons in component', () => {
    const mockClose = jest.fn();
    const mockSwitch = jest.fn();
    const mockSubmit = jest.fn();

    const { getByTestId } = render(
      <CreatePickDialog
        selectedSection={MOVE_TO_FRONT}
        setSelectedSection={jest.fn()}
        numberOfPallets={1}
        setNumberOfPallets={jest.fn()}
        isQuickPick={false}
        setIsQuickPick={mockSwitch}
        onClose={mockClose}
        onSubmit={mockSubmit}
        locations={[]}
      />,
    );

    const closeButton = getByTestId('closeButton');
    const switchButton = getByTestId('quickPickSwitch');
    const submitButton = getByTestId('submitButton');
    fireEvent.press(closeButton);
    expect(mockClose).toHaveBeenCalled();
    fireEvent(switchButton, 'onValueChange', true);
    expect(mockSwitch).toHaveBeenCalled();
    fireEvent.press(submitButton);
    expect(mockSubmit).toHaveBeenCalled();
  });
  it('should test existence of noOfpallets input element based on selectedSection other than MOVE_TO_FRONT', () => {
    const { queryAllByText } = render(
      <CreatePickDialog
        selectedSection="ABAR1-1"
        setSelectedSection={jest.fn()}
        numberOfPallets={1}
        setNumberOfPallets={jest.fn()}
        isQuickPick={false}
        setIsQuickPick={jest.fn}
        onClose={jest.fn}
        onSubmit={jest.fn}
        locations={[]}
      />,
    );
    expect(queryAllByText(strings('PICKING.NUMBER_PALLETS'))).toHaveLength(1);
  });
});
