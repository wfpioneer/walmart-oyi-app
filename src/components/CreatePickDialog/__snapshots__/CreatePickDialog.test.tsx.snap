// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test createPickDialog Component Renders the createPickDialog component with location moveToFront selected 1`] = `
<View>
  <View
    style={
      Object {
        "alignContent": "flex-start",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <IconButton
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      icon={
        <mockMaterialCommunityIcons
          color="#9e9e9e"
          name="close"
          size={16}
        />
      }
      onPress={[Function]}
      radius={5}
      testID="closeButton"
      type={3}
      width={-1}
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    [missing "en.PICKING.SELECT_LOCATION" translation]
  </Text>
  <View
    style={
      Object {
        "borderColor": "#c5c5c5",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginHorizontal": 20,
      }
    }
  >
    <Picker
      mode="dropdown"
      onValueChange={[Function]}
      selectedValue="moveToFront"
    >
      <PickerItem
        label="[missing \\"en.PICKING.SELECT_LOCATION\\" translation]"
        value=""
      />
      <PickerItem
        label="[missing \\"en.PICKING.MOVE_TO_FRONT\\" translation]"
        value="moveToFront"
      />
    </Picker>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    <Text>
      [missing "en.PICKING.NUMBER_PALLETS" translation]
    </Text>
    <NumericSelector
      isValid={true}
      maxValue={99}
      minValue={1}
      onDecreaseQty={[Function]}
      onIncreaseQty={[Function]}
      onTextChange={[Function]}
      value={1}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      [missing "en.PICKING.QUICK_PICK" translation]
    </Text>
    <Switch
      onValueChange={[Function]}
      testID="quickPickSwitch"
      thumbColor="#0071e9"
      trackColor={
        Object {
          "false": "#c5c5c5",
          "true": "#0071e9",
        }
      }
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 20,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[MockFunction]}
      radius={5}
      testID="submitButton"
      title="[missing \\"en.PICKING.CREATE_PICK\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</View>
`;

exports[`Test createPickDialog Component Renders the createPickDialog component with location other than moveToFront selected 1`] = `
<View>
  <View
    style={
      Object {
        "alignContent": "flex-start",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <IconButton
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      icon={
        <mockMaterialCommunityIcons
          color="#9e9e9e"
          name="close"
          size={16}
        />
      }
      onPress={[Function]}
      radius={5}
      testID="closeButton"
      type={3}
      width={-1}
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    [missing "en.PICKING.SELECT_LOCATION" translation]
  </Text>
  <View
    style={
      Object {
        "borderColor": "#c5c5c5",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginHorizontal": 20,
      }
    }
  >
    <Picker
      mode="dropdown"
      onValueChange={[Function]}
      selectedValue=""
    >
      <PickerItem
        label="[missing \\"en.PICKING.SELECT_LOCATION\\" translation]"
        value=""
      />
      <PickerItem
        label="[missing \\"en.PICKING.MOVE_TO_FRONT\\" translation]"
        value="moveToFront"
      />
    </Picker>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      [missing "en.PICKING.QUICK_PICK" translation]
    </Text>
    <Switch
      onValueChange={[Function]}
      testID="quickPickSwitch"
      thumbColor="#0071e9"
      trackColor={
        Object {
          "false": "#c5c5c5",
          "true": "#0071e9",
        }
      }
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 20,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[MockFunction]}
      radius={5}
      testID="submitButton"
      title="[missing \\"en.PICKING.CREATE_PICK\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</View>
`;
